sktime/utils/plotting.py

```diff
@@ -64,13 +62,21 @@ def plot_series(*series, labels=None, markers=None):
     else:
         markers = ["o" for _ in range(n_series)]
 
+    def _type(id):
+        return str(type(id)).split(".")[-1].split("'")[0]
+
     # create combined index
     index = series[0].index
+    idx_type = type(index)
     for y in series[1:]:
         # check types, note that isinstance() does not work here because index
         # types inherit from each other, hence we check for type equality
+        acc_dtypes = ["RangeIndex", "Int64Index", "UInt64Index", "Float64Index"]
         if not type(index) is type(y.index):  # noqa
```

Comment:
I would try something like this: 
```python
def _is_int_index(index):
    return type(index) in (pd.Int64Index, pd.RangeIndex)


def _check_consistent_index_type(*series):
    
    index = series[0].index    
    for y in series[1:]: 
        
        msg = "Found `series` with inconsistent index types. Please make sure that all series have the same index type."
        
        if _is_int_index(y.index):
            if not _is_int_index(index):
                raise TypeError(msg)
                
        else:
            if type(index) is not type(y.index):
                raise TypeError(msg)
```

We can then use the function `_check_consistent_index_type` at the beginning of the `plot_series` function. What do you think?

Reply:
Yes, this makes complete sense. I realised that the 'union' method includes implicit type casting, so there should be no need for manually doing so in the code now.